identity/app/controllers/editprofile/ConsentsJourney.scala

```diff
@@ -91,7 +91,12 @@ trait ConsentsJourney
     consentHint: Option[String]): Action[AnyContent] =
     csrfAddToken {
       consentsRedirectAction.async { implicit request =>
-        val returnUrl = returnUrlVerifier.getVerifiedReturnUrl(request).getOrElse(returnUrlVerifier.defaultReturnUrl)
+
+        val returnUrl = returnUrlVerifier.getVerifiedReturnUrl(request) match {
+          case Some(url) => if (url contains "/consents") returnUrlVerifier.defaultReturnUrl else url
```

Comment:
Massively late to the party but I would probably put the `if` as an actual guard on the `case` and have a third `case`! Next time!